programmers/소수-찾기.js

```diff
@@ -0,0 +1,41 @@
+function combine(splitedNums, N) {
+    let result = [];
+    if (N === 1) return splitedNums.map((v) => [v]);
+  
+    splitedNums.forEach((v, idx, splitedNums) => {
+      const fixer = v;
+      const rests = splitedNums.filter((_, index) => index !== idx);
+      const permuationArr = combine(rests, N - 1);
+      const combineFixer = permuationArr.map((v) => [fixer, ...v]);
+      result.push(...combineFixer);
+    });
```

Comment:
변수 shadowing 하지말기
[no-shadow](https://eslint.org/docs/rules/no-shadow)
```suggestion
    splitedNums.forEach((v, idx, arr) => {
      const fixer = v;
      const rests = arr.filter((_, index) => index !== idx);
      const permuationArr = combine(rests, N - 1);
      const combineFixer = permuationArr.map((val) => [fixer, ...val]);
      result.push(...combineFixer);
    });
```